src/utils/fromJS.js

```diff
@@ -8,6 +8,35 @@
 
 const Immutable = require("immutable");
 
+/*
+  creates an immutable map, where each of the value's
+  items are transformed into their own map.
+
+  NOTE: we guard against `length` being a property because
+  length confuses Immutable's internal algorithm.
+*/
+function createMap(value) {
+  const hasLength = value.hasOwnProperty("length");
+  const length = value.length;
+
+  if (hasLength) {
+    value.length = `${ value.length}`;
```

Comment:
nit: space between `${` and `value.length}` and not between `value.length` and `}`